pure.zsh

```diff
@@ -390,6 +390,10 @@ prompt_pure_async_callback() {
 	local do_render=0
 
 	case $job in
+		\[async])
+			# our worker died unexpectedly
+			typeset -g prompt_pure_async_init=0
```

Comment:
The worker could have crashed during some async invocation after the prompt has rendered. In this case, just setting init to zero will not recreate the worker and re-initiate the tasks. Not sure if we want to handle it or not, but I think we do. Worst case the user is left with an empty prompt until they hit enter (e.g. missing git information).

Reply:
I'd say this is not a problem. Rationale: pure is async as in, if information is currently unavailable, then bail out. Here is a new case where information is unavailable, so bail out. On the next prompt draw, we will retry. This seems sane to me.

Also, this keeps complexity low. I am afraid doing better would be fairly complex for a scenario which is very infrequent at best.